### PR TITLE
[bugfix] Prevent Automatic Topic Creation during namespace deletion

### PR DESCRIPTION
**Pull request by eolivelli**
### Motivation

During the deletion of a namespace we don't want that autoTopicCreation triggers the creation of topics.
This is related to #17308 but it deals only with Namespace and Tenant deletion.

### Modifications

Consider "Policies.deleted" while deciding if autoTopicCreation is available or not

### Verifying this change
I will add tests

### Documentation

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)

- [x] `doc-not-needed` 
(Please explain why)

- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Comment by eolivelli** [comment]
@poorbarcode @dlg99 PTAL

**Comment by eolivelli** [comment]
@poorbarcode unfortunately my test case doesn't cover well the change, because basically the automatic topic creation is prevented earlier, because we have the check about policies.deleted in PulsarWebResources.checkLocalOrGetPeerReplicationCluster() 

do you have some suggestions ?

**Review by poorbarcode — APPROVED** [review]
This reduces the probability of problems occurring. When the namespace deletion instruction is not processed by the current broker,  may succeed in creating the topic because the meta data has a brief cache.

**Comment by eolivelli** [comment]
> When the namespace deletion instruction is not processed by the current broker, may succeed in creating the topic because the meta data has a brief cache.

This is a good point, but I would defer it to the most bigger issue of cache consistency

**Comment by eolivelli** [comment]
@poorbarcode I added another test that demonstrates that this patch is covering one edge case that you mentioned.
We must prevent auto topic creation not only during lookup but also while creating a consumer or a producer.

The new test covers this change now

**Review by dlg99 — APPROVED** [review]
LGTM

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/TopicAutoCreationTest.java [hunk 1]
```diff
@@ -85,4 +105,76 @@ public void testPartitionedTopicAutoCreation() throws PulsarAdminException, Puls
...
+                return CompletableFuture.completedFuture(Pair.of(brokerAddress, brokerAddress));
+            });
+
+            // Creating a producer and creating a Consumer may trigger automatic topic
+            // creation, let's try to create a Producer and a Consumer
+            try (Producer<byte[]> producer = pulsarClient.newProducer()
+                    .sendTimeout(1, TimeUnit.SECONDS)
+                    .topic(topic)
+                    .create();) {
+            } catch (PulsarClientException.LookupException expected) {
```
Comment (poorbarcode) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/TopicAutoCreationTest.java:145]: Should we add validation to make sure the exception is what we expect? When I run this test, the exception is "Namespace bundle for topic (%s) not served by this instance" which should not be what we expected.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/TopicAutoCreationTest.java [hunk 2]
```diff
@@ -85,4 +105,76 @@ public void testPartitionedTopicAutoCreation() throws PulsarAdminException, Puls
...
+            }
+
+            try (Consumer<byte[]> consumer = pulsarClient.newConsumer()
+                    .topic(topic)
+                    .subscriptionName("test")
+                    .subscribe();) {
+            } catch (PulsarClientException.LookupException expected) {
```
Comment (poorbarcode) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/TopicAutoCreationTest.java:152]: Same as above

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/TopicAutoCreationTest.java (hunk 1 above)
Comment (eolivelli) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/TopicAutoCreationTest.java:145]: This is happening because the broker hasn't loaded the bundle.
because basically the producer/consumer could not set them up.

I wasn't sure to add an assertion about "Namespace bundle for topic (%s) not served by this instance"

btw maybe it is better, so that the behaviour changes we will see this test failing

Comment (eolivelli) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/TopicAutoCreationTest.java:146]: done

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/TopicAutoCreationTest.java (hunk 2 above)
Comment (eolivelli) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/TopicAutoCreationTest.java:156]: done

**Review by codelipenghui — APPROVED** [review]
